### PR TITLE
Fixes blob diagonal expansion

### PR DESCRIPTION
**Pull request by chuga-git**
## What Does This PR Do

Fixes diagonal expansion sometimes breaking when the blob tile does not have an assigned overmind. I don't know why blob tiles sometimes don't get assigned an overmind when `Pulse()` is called. They just do. This antag needs serious help.
## Why It's Good For The Game

It was broken (sometimes) and now it's not

## Testing

Didn't runtime. I can't check how the overmind assignment works with a second overmind and there is a small chance something wonky will happen.
## Changelog
:cl: Chuga
fix: Made blob diagonal expansion less buggy
/:cl:

### CODE REVIEW

File: code/modules/events/blob/theblob.dm [hunk 1]
```diff
@@ -116,26 +119,24 @@ GLOBAL_LIST_EMPTY(blob_minions)
...
-		if(!B)
-			expand(T, 1, a_color, overmind)//No blob here so try and expand
-			return
-		B.adjustcolors(a_color)
+	if(!length(dirs))
+		return
 
-		B.Pulse((pulse+1),get_dir(src.loc,T), a_color)
+	var/dirn = pick(dirs)
+	dirs.Remove(dirn)
```
Comment (Contrabang) [inline code/modules/events/blob/theblob.dm:128]:
```suggestion
	var/dirn = pick_n_take(dirs)
```

File: code/modules/events/blob/theblob.dm [hunk 2]
```diff
@@ -116,26 +119,24 @@ GLOBAL_LIST_EMPTY(blob_minions)
...
+	var/turf/T = get_step(src, dirn)
+	var/obj/structure/blob/B = (locate(/obj/structure/blob) in T)
```
Comment (Contrabang) [inline code/modules/events/blob/theblob.dm:130]:
```suggestion
	var/obj/structure/blob/B = locate(/obj/structure/blob) in T
```

File: code/modules/events/blob/theblob.dm [hunk 3]
```diff
@@ -116,26 +119,24 @@ GLOBAL_LIST_EMPTY(blob_minions)
...
+	if(!B)
+		expand(T, 1, a_color, overmind)	//No blob here so try and expand
 		return
-	return
 
+	B.adjustcolors(a_color)
+	B.Pulse((pulse+1),get_dir(src.loc,T), a_color)
```
Comment (Contrabang) [inline code/modules/events/blob/theblob.dm:136]:
```suggestion
	B.Pulse(pulse+1, get_dir(loc, T), a_color)
```

**Review by Contrabang — CHANGES_REQUESTED** [review]

File: code/modules/events/blob/theblob.dm [hunk 4]
```diff
@@ -49,10 +49,10 @@ GLOBAL_LIST_EMPTY(blob_minions)
 
 /obj/structure/blob/CanPass(atom/movable/mover, turf/target, height=0)
 	if(height==0)
```
Comment (DGamerL) [inline code/modules/events/blob/theblob.dm:51]:
```suggestion
	if(height == 0)
```

File: code/modules/events/blob/theblob.dm [hunk 5]
```diff
@@ -49,10 +49,10 @@ GLOBAL_LIST_EMPTY(blob_minions)
...
-		return 1
+		return TRUE
 	if(istype(mover) && mover.checkpass(PASSBLOB))
```
Comment (DGamerL) [inline code/modules/events/blob/theblob.dm:53]:
```suggestion
	return istype(mover) && mover.checkpass(PASSBLOB)
```

File: code/modules/events/blob/theblob.dm [hunk 6]
```diff
@@ -49,10 +49,10 @@ GLOBAL_LIST_EMPTY(blob_minions)
...
-		return 1
-	return 0
+		return TRUE
+	return FALSE
```
Comment (DGamerL) [inline code/modules/events/blob/theblob.dm:55]:
```suggestion
```
Github wasn't allowing me to do both of these in one comment

File: code/modules/events/blob/theblob.dm [hunk 7]
```diff
@@ -146,28 +146,40 @@ GLOBAL_LIST_EMPTY(blob_minions)
 /obj/structure/blob/proc/expand(turf/T = null, prob = 1, a_color, _overmind = null, turf/double_target = null)
```
Comment (DGamerL) [inline code/modules/events/blob/theblob.dm:146]:
```suggestion
/obj/structure/blob/proc/expand(turf/T, prob = TRUE, a_color, _overmind, turf/double_target)
```

File: code/modules/events/blob/theblob.dm [hunk 8]
```diff
@@ -146,28 +146,40 @@ GLOBAL_LIST_EMPTY(blob_minions)
...
 	if(prob && !prob(obj_integrity))
 		return
-	if(isspaceturf(T) && prob(75)) 	return
+
+	if(isspaceturf(T) && prob(75))
+		return
+
 	if(!T)
 		var/list/dirs = list(1,2,4,8)
```
Comment (DGamerL) [inline code/modules/events/blob/theblob.dm:154]:
```suggestion
		var/list/dirs = list(1, 2, 4, 8)
```

File: code/modules/events/blob/theblob.dm [hunk 9]
```diff
@@ -183,14 +195,14 @@ GLOBAL_LIST_EMPTY(blob_minions)
 			overmind.blob_reagent_datum.reaction_mob(M, REAGENT_TOUCH, 25, 1, mob_protection)
 			overmind.blob_reagent_datum.send_message(M)
 		A.blob_act(src)
-	return 1
+	return TRUE
 
-/obj/structure/blob/proc/double_expand(turf/T = null, prob = 1, a_color, _overmind = null)
+/obj/structure/blob/proc/double_expand(turf/T = null, prob = 1, a_color, mob/camera/blob/incoming_overmind = null)
```
Comment (DGamerL) [inline code/modules/events/blob/theblob.dm:200]:
```suggestion
/obj/structure/blob/proc/double_expand(turf/T, prob = TRUE, a_color, mob/camera/blob/incoming_overmind)
```

**Review by DGamerL — APPROVED** [review]

File: code/modules/events/blob/theblob.dm [hunk 10]
```diff
@@ -116,58 +117,66 @@ GLOBAL_LIST_EMPTY(blob_minions)
...
+	var/dirn = pick_n_take(dirs)
+	var/turf/T = get_step(src, dirn)
+	var/obj/structure/blob/B = locate(/obj/structure/blob) in T
+	if(!B)
+		expand(T, 1, a_color, overmind)	//No blob here so try and expand
 		return
-	return
 
+	B.adjustcolors(a_color)
+	B.Pulse(pulse+1, get_dir(loc, T), a_color)
```
Comment (DGamerL) [inline code/modules/events/blob/theblob.dm:133]:
```suggestion
	B.Pulse(pulse + 1, get_dir(loc, T), a_color)
```

**Review by DGamerL — APPROVED** [review]
LGTM

**Review by Sirryan2002 — APPROVED** [review]

**Review by Contrabang — APPROVED** [review]
Send it

**Review by DGamerL — APPROVED** [review]
Send it

**Pull request merged**